### ISSUE TITLE
Glue 4.0 only succeeds first run

### ISSUE THREAD

**Issue by rickiesmooth** [aws-samples/dbt-glue#228]
### Describe the bug
Glue doesn't recognize existing delta table produced by `dbt-glue`

### Steps To Reproduce
* Configure a dbt-glue project using `datalake_formats: delta` , eg:
```yml
demo:
  target: dev
  outputs:
    dev:
      type: glue
      query-comment: demo
      role_arn: "{{ env_var('DBT_ROLE_ARN') }}"
      region: us-east-1
      glue_version: "4.0"
      workers: 2
      worker_type: G.1X
      schema: "demo"
      database: "demo"
      session_provisioning_timeout_in_seconds: 120
      location: "{{ env_var('DBT_S3_LOCATION') }}"
      datalake_formats: delta
      conf: "spark.sql.extensions=io.delta.sql.DeltaSparkSessionExtension --conf spark.sql.catalog.spark_catalog=org.apache.spark.sql.delta.catalog.DeltaCatalog"
      extra_py_files: "/opt/aws_glue_connectors/selected/datalake/delta-core_2.12-2.1.0.jar"
```

```sql
{{ config(
    incremental_strategy='insert_overwrite',
    partition_by=['team_id', 'date_day'],
    file_format='delta',
) }}

with new_events as (

    select * from {{ ref('events') }}

    {% if is_incremental() %}
    where date_day >= date_add(current_date, -1)
    {% endif %}

)

select
    user_id,
    max(date_day) as last_seen,
    current_date() as dt

from events
group by 1
```

* Run dbt-glue project to create the initial delta table
* Run dbt-glue project after initial delta table is created and it returns the following error:
```bash
07:23:29      '''), AnalysisException: Cannot create table ('`demo`.`silver_pageviews`'). The associated location ('s3://prod-pipeline-data-lake-output/demo/silver_pageviews') is not empty but it's not a Delta table
```
* navigate to `s3://prod-pipeline-data-lake-output/demo/silver_pageviews` and observe there's a `_delta_log` directory in the location.

### Expected behavior
Glue recognizes associated location as delta table

### System information
**The output of `dbt --version`:**
```
07:20:35  Running with dbt=1.6.1
07:[20](https://github.com/rickiesmooth/trackagree/actions/runs/6006491699/job/16298048718#step:6:21):35  Registered adapter: glue=1.6.2
```

**The operating system you're using:**

**The output of `python --version`:**

**Comment by rickiesmooth** [comment]
Am I doing something wrong? When I use parquet I get a similar error

```sql
{{ config(
    incremental_strategy='insert_overwrite',
    partition_by=['team_id', 'date_day'],
    file_format='parquet',
) }}
```

```bash
08:51:18      '''), AnalysisException: CREATE-TABLE-AS-SELECT cannot create table with location to a non-empty directory s3://prod-pipeline-data-lake-output/demo/silver_pageviews . To allow overwriting the existing non-empty directory, set 'spark.sql.legacy.allowNonEmptyLocationInCTAS' to true.
```

**Comment by rickiesmooth** [comment]
it seems to do a create, while I expect it would do a create or replace:
```
15:32:00  Database Error in model silver_pageviews (models/silver/silver_pageviews.sql)
15:32:00    Glue cursor returned `error` for statement None for code SqlWrapper2.execute('''/* {"app": "dbt", "dbt_version": "1.6.1", "profile_name": "demo", "target_name": "dev", "node_id": "model.demo.silver_pageviews"} */
15:32:00    
15:32:00      
15:32:00        create table trackagree.silver_pageviews
15:32:00      
15:32:00        using delta
15:32:00    	partitioned by (team_id,date_day)
15:32:00    	
15:32:00    	
15:32:00    	LOCATION 's3://trackagree-prod-pipeline-data-lake-output/demo/silver_pageviews/'
15:32:00    	
15:32:00    	as
```

**Comment by rickiesmooth** [comment]
@menuetb can you maybe spot what I'm doing wrong? Seems like a pretty standard setup

**Comment by rickiesmooth** [comment]
ah downgrading to 3.0 fixes the issue

**Comment by vitoravancini** [comment]
why was this closed? dbt glue only works with glue 3.0 for parquet?

I'm having the same issue when running dbt run for the second time.
The error points to a config (spark.sql.legacy.allowNonEmptyLocationInCTAS) but setting it in the sessions does not seem to change anything.

Anyone having success with dbt glue and parquet?

**Comment by vilhomaa** [comment]
I am still experiencing this same issue
